### PR TITLE
Eng 1608 fix

### PR DESCRIPTION
**Pull request by glenntownsend**
The SIGTERM signal results in some wonky behavior in rules_pyz
After testing, I found out I had to put this here AND in algo to get it to work all the time.

### CODE REVIEW

File: rules_python_zip/simplepack.go [hunk 1]
```diff
@@ -501,6 +502,14 @@ if need_unzip and isinstance(__loader__, zipimport.zipimporter):
     tempdir_create_pid = os.getpid()
     atexit.register(clean_tempdir_parent_only, tempdir)
     sys.path.insert(0, tempdir)
+    # Handle linux signal terminate by calling exit, so atexit code executes.
+    old_handler = None
+    def sig_exit(*args):
+        if sys.path[0].endswith('_pyzip'):
```
Comment (joshclimacell) [inline rules_python_zip/simplepack.go]: The temp dir is stored in the `tempdir` variable just above. Why not use that...

File: rules_python_zip/simplepack.go [hunk 2]
```diff
@@ -501,6 +502,14 @@ if need_unzip and isinstance(__loader__, zipimport.zipimporter):
     tempdir_create_pid = os.getpid()
     atexit.register(clean_tempdir_parent_only, tempdir)
     sys.path.insert(0, tempdir)
+    # Handle linux signal terminate by calling exit, so atexit code executes.
```
Comment (joshclimacell) [inline rules_python_zip/simplepack.go]: Better documentation please. Explain why this signal stuff is needed... And the signal-handling code you're adding does not cause anything to do with `atexit` to happen, so I don't think this is correct...

File: rules_python_zip/simplepack.go [hunk 3]
```diff
@@ -501,6 +502,14 @@ if need_unzip and isinstance(__loader__, zipimport.zipimporter):
...
+    old_handler = None
+    def sig_exit(*args):
+        if sys.path[0].endswith('_pyzip'):
+            shutil.rmtree(sys.path[0])
```
Comment (joshclimacell) [inline rules_python_zip/simplepack.go]: Duplicate `shutil` call -- why not just use the existing `clean_tempdir_parent_only` function... It handles the possibility of forking, which I don't believe your code does?

File: rules_python_zip/simplepack.go [hunk 4]
```diff
@@ -501,6 +502,14 @@ if need_unzip and isinstance(__loader__, zipimport.zipimporter):
...
+        if old_handler:
+            old_handler(*args)
+    old_handler = signal.signal(signal.SIGTERM, sig_exit)
```
Comment (joshclimacell) [inline rules_python_zip/simplepack.go]: I thought we decided having pyz code provide signal handlers was a bad idea, given that it's a reasonably common thing for code that might get packaged into a pyz binary to want to do.

File: rules_python_zip/simplepack.go (hunk 2 above)
Comment (joshclimacell) [inline rules_python_zip/simplepack.go]: If we do end up putting this in, can you please send an equivalent PR to the main pyz repo? I don't want to diverge any further, given that we will very likely want to merge in upstream changes in the near future.

**Review by joshclimacell — CHANGES_REQUESTED** [review]

File: rules_python_zip/simplepack.go (hunk 2 above)
Comment (glenntownsend) [reply rules_python_zip/simplepack.go]: I've updated the documentation to be explicit on the why.

Comment (glenntownsend) [reply rules_python_zip/simplepack.go]: I can do that, it's not a big change.

File: rules_python_zip/simplepack.go (hunk 1 above)
Comment (glenntownsend) [reply rules_python_zip/simplepack.go]: Changed to that, as it works correctly and is clearer. 
(had to try many things to get it all working).

File: rules_python_zip/simplepack.go (hunk 3 above)
Comment (glenntownsend) [reply rules_python_zip/simplepack.go]: Yup, that works too. Updated.

File: rules_python_zip/simplepack.go (hunk 4 above)
Comment (glenntownsend) [reply rules_python_zip/simplepack.go]: That was the plan, but while it works properly for the lightning forecast (what I was testing it on originally) it failed to work properly for any of the multi-threaded tools I've got. Had to test a bunch of permutations to figure out how to get it working correctly.

**Review by glenntownsend — COMMENTED** [review]
ping @joshclimacell

File: rules_python_zip/simplepack.go (hunk 4 above)
Comment (joshclimacell) [reply rules_python_zip/simplepack.go]: Hmm, could you elaborate? Why did it work in some cases without modifying pyz, but not in others? In what way did it not work? Could it be this forking issue? Are you using multiprocessing? The code here LGTM, but I'd favor doing signal handling elsewhere unless this is the only way (which seems unlikely...).

Comment (joshclimacell) [reply rules_python_zip/simplepack.go]: (Or happy to chat in person)

**Comment by glenntownsend** [comment]
ping @joshclimacell an in-person meeting's not going to happen for a while. We could use zoom?
As for why I specifically need to have the signal handling in this code, the short answer is 'I don't know'. I figured out HOW to get the folders deleting by experimentation rather than trying to delve the code because I'm pressed for time ATM.  I know it works this way, and doesn't work if I don't catch that signal.

**Review by joshclimacell — APPROVED** [review]
LGTM

**Pull request merged**